src/storage/txn/commands/flashback_to_version_read_phase.rs

```diff
@@ -147,31 +146,45 @@ impl<S: Snapshot> ReadCommand<S> for FlashbackToVersionReadPhase {
                     }
                 }
             }
-            FlashbackToVersionState::FlashbackWrite { next_write_key, .. } => {
+            FlashbackToVersionState::FlashbackWrite {
+                mut next_write_key, ..
+            } => {
                 if self.commit_ts <= self.start_ts {
                     return Err(Error::from(ErrorInner::InvalidTxnTso {
                         start_ts: self.start_ts,
                         commit_ts: self.commit_ts,
                     }));
                 }
-                // If the key is not locked, it means that the key has been committed before and
-                // we are in a retry.
-                if next_write_key == self.start_key && reader.load_lock(&next_write_key)?.is_none()
-                {
-                    return Ok(ProcessResult::Res);
+                if next_write_key == self.start_key {
+                    // Get real key first.
+                    let key_to_lock = if let Some(first_key) =
+                        get_first_user_key(&mut reader, &self.start_key, &self.end_key)?
+                    {
+                        first_key
+                    } else {
+                        statistics.add(&reader.statistics);
+                        return Ok(ProcessResult::Res);
+                    };
+                    next_write_key = key_to_lock.clone();
+                    start_key = key_to_lock;
```

Comment:
This could eliminate the `clone`.

```suggestion
                    let next_write_key = if let Some(first_key) =
                        get_first_user_key(&mut reader, &self.start_key, &self.end_key)?
                    {
                        first_key
                    } else {
                        statistics.add(&reader.statistics);
                        return Ok(ProcessResult::Res);
                    };
                    start_key = key_to_lock;
```